frontend/storage/configureStore.ts

```diff
@@ -0,0 +1,10 @@
+// eslint-disable-next-line no-unused-vars
+import {createStore, Store} from 'redux';
+// eslint-disable-next-line no-unused-vars
+import { Breadcrumb, SetBreadcrumbAction } from './common';
+import { breadcrumbReducer } from './reducers/breadcrumb';
+
+export default function configureStore(): Store<Breadcrumb, SetBreadcrumbAction> {
+	const store = createStore(breadcrumbReducer);
```

Comment:
Ah, okay, I see why you have the `Breadcrumb` interface in the first place, because it's the interface for the whole store.  Okay, let me talk a bit more about Redux. I'll probably duplicate a bit of what you already know.

1. Redux works best if you have only a single store. If you don't want to have a single enormous reducer handling everything there are a bunch of ways to break it down, but my preferred one is nested reducers, like i have in that private project I shared with you:

```
export function mapViewerDataReducer(
	state: GameDataState,
	action: MapViewerDataAction,
): GameDataState {
	state = gameReducer(state, action);
	state = mapReducer(state, action);
	state = areaReducer(state, action);
	state = roomReducer(state, action);
	state = roomTileReducer(state, action);

	return state;
}
```

2. Since you want to have everything nested under a single object, the outermost layer should be generic and reflect the name of the state as a whole, so `AppState` or `AppStore` would be a good fit.

3. Redux figures out when to rerender things using identity comparison `===`, so for things that did not change you ideally want to change nothing.

 -----

**Actionables:** I'll likely duplicate things from other comments. Let's aim for simplicity now:
1. Rename `Breadcrumb` to something more appropriate for the data of the whole application - `AppState` or, to avoid mixing things up with components' internal state, `AppStore` 
2. Accordingly rename all instances of `Breadcrumb` to the proper "global" name.
3. In the state and reducer itself change `pageName` to `breadcrumb` or `headerBreadcrumb`
4. Keep the `SET_BREADCRUMB` and `SetBreadcrumbAction` names, since those fit as they only change the breadcrumb.